bundler/lib/bundler/compact_index_client/updater.rb

```diff
@@ -99,8 +100,27 @@ def checksum_for_file(path)
         end
       end
 
+      def sha256_digest_for_file(path)
+        return nil unless path.file?
+        SharedHelpers.filesystem_access(path, :read) do
+          Base64.strict_encode64(SharedHelpers.digest(:SHA256).digest(File.read(path)))
+        end
+      end
+
       private
 
+      def sha256_from_digest(digest)
+        return nil if digest.nil?
+
+        digest&.split(",")&.find {|s| s.start_with?("sha-256=") }&.delete_prefix("sha-256=")&.delete('"')
```

Comment:
Maybe a bit more memory efficient.

```suggestion
        sha256 = digest[/sha-256=[^,]+/]
        return unless sha256
        sha256.delete_prefix!("sha-256=")
        sha256.delete!('":') # All the RFC examples for Digest use `:` as the boundary character.
        sha256
```